flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandlerTest.java

```diff
@@ -430,23 +431,35 @@ private static void releaseResource(SingleInputGate inputGate, NetworkBufferPool
 	/**
 	 * Returns a deserialized buffer message as it would be received during runtime.
 	 */
-	private static BufferResponse createBufferResponse(
+	private BufferResponse createBufferResponse(
 			Buffer buffer,
 			int sequenceNumber,
-			InputChannelID receivingChannelId,
-			int backlog) throws IOException {
+			RemoteInputChannel receivingChannel,
+			int backlog,
+			CreditBasedPartitionRequestClientHandler clientHandler) throws IOException {
+
 		// Mock buffer to serialize
-		BufferResponse resp = new BufferResponse(buffer, sequenceNumber, receivingChannelId, backlog);
+		BufferResponse resp = new BufferResponse(
+				buffer,
+				sequenceNumber,
+				receivingChannel.getInputChannelId(),
+				backlog);
 
 		ByteBuf serialized = resp.write(UnpooledByteBufAllocator.DEFAULT);
 
 		// Skip general header bytes
 		serialized.readBytes(NettyMessage.FRAME_HEADER_LENGTH);
 
+
 		// Deserialize the bytes again. We have to go this way, because we only partly deserialize
 		// the header of the response and wait for a buffer from the buffer pool to copy the payload
 		// data into.
-		BufferResponse deserialized = BufferResponse.readFrom(serialized);
+		NetworkBufferAllocator allocator = new NetworkBufferAllocator(clientHandler);
```

Comment:
It seems better to pass `NetworkBufferAllocator` in the argument. Assuming we create multiple `BufferResponse` in one test, it is not necessary to create allocator for every `BufferResponse`.

Reply:
Changed the parameter to allocator.